core/src/main/java/org/fao/geonet/api/records/attachments/AbstractStore.java

```diff
@@ -232,6 +232,13 @@ public final MetadataResource putResource(ServiceContext context, String metadat
         if (contentDisposition != null) {
             filename = ContentDisposition.parse(contentDisposition).getFilename();
         }
+        // If follow redirect, get the filename from the redirected URL
+        if (filename == null && connection.getInstanceFollowRedirects()) {
```

Comment:
Some questions:

1) Should not check also for empty value, like in line 242?
2) Should not change line 242 to use `else if ....` ?